### DISCUSSION TITLE
`AStarGrid2D.is_point_solid` should be renamed to `get_point_solid`

### DISCUSSION THREAD

**Discussion by hunterloftis** [godotengine/godot-proposals#9358] [Engine Core]
"Set"ters should be paired with "Get"ters. Much like it already has the pair of `set_point_weight_scale` and `get_point_weight_scale`, it should have `set_point_solid` and `get_point_solid`.

**Comment by AThousandShips** [comment]
This would break compatibility, further the `set/is` pattern is well established for Boolean properties, so nothing wrong here, see for example:
* [`CanvasItem.visible`](https://docs.godotengine.org/en/stable/classes/class_canvasitem.html#class-canvasitem-property-visible)
* [`Node3D.top_level`](https://docs.godotengine.org/en/stable/classes/class_node3d.html#class-node3d-property-top-level)
* [`Resource.resource_local_to_scene`](https://docs.godotengine.org/en/stable/classes/class_resource.html#class-resource-property-resource-local-to-scene)
* Etc.

This would *break* the pattern with `is_jumping_enabled`

While there *are* cases of `get` with boolean properties they are far, *far* rarer, just a casual search shows that `is` is far more common, there's only a few hundred cases of the former, but thousands of the latter, that is in the c++ side of the code, not all of them represent property getters, but it shows the scale, there's also `has` for the same pattern

When searching the bound properties the following is clear:
* `get`: 297
* `is`: 530
* `has`: 15
* `are`: 8

**Comment by hunterloftis** [comment]
Fair enough. The lack of symmetry is surprising to me, but maybe that's because Godot isn't my normal environment.

If it's a well-established pattern in Godot, it isn't a consistent one (for example [Bitmap's](https://docs.godotengine.org/en/stable/classes/class_bitmap.html) set_bit / get_bit and set_bitv / get_bitv).

**Reply by AThousandShips** [reply]
There's no lack of symmetry though? Only if you define "get" the complement to "set", it depends on how you interpret it, so it's just down to assumptions being made

for `get_bitv` it doesn't fit, a `bit` isn't a truth value in the same sense I'd say, but as you can see it's a *very* well established one, with only 297 out of 850 being of the `get` form with properties

For methods it's even stronger:
* `get`: 84
* `is`: 372
* `has`: 139
* `are`: 2

**Reply by hunterloftis** [reply]
I've already closed the issue as it's not worth a bikeshed, was just a surprise to me coming into Godot from other environments. However, I'm not sure I would agree that 65% consistency is a "very well" established pattern in API design.

**Reply by AThousandShips** [reply]
Well it's a well established design pattern that's largely followed, with new additions being used for it, and it's well established among the maintainers :) These are exceptions, by accident or by compatibility
